packages/esm-care-panel-app/src/dsdm-history/dsdm-history.component.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo, useState } from 'react';
+import {
+  DataTable,
+  DataTableSkeleton,
+  Pagination,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableToolbar,
+  Layer,
+  Tile,
+} from '@carbon/react';
+
+import { useTranslation } from 'react-i18next';
+import { formatDate, parseDate, usePagination } from '@openmrs/esm-framework';
+import styles from './dsdm-history.scss';
+import { usePatientPrograms } from '../hooks/usePatientPrograms';
+
+interface dsdmnProps {
+  patientUuid: string;
+}
+
+const DSDMHistory: React.FC<dsdmnProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+
+  const { isLoading, error, dsdmModels } = usePatientPrograms(patientUuid);
+
+  const pageSizes = [10, 20, 30, 40, 50];
+  const [currentPageSize, setPageSize] = useState(10);
+
+  const { goTo, results: paginatedDSDM, currentPage } = usePagination(dsdmModels, currentPageSize);
+  let columns = [
+    { id: 0, header: t('model', 'Model'), key: 'model' },
+
+    { id: 1, header: t('dateEnrolled', 'Date Enrolled'), key: 'dateEnrolled' },
+    { id: 2, header: t('dateCompleted', 'Date Completed'), key: 'dateCompleted' },
+  ];
+
+  const tableRows = useMemo(() => {
+    const currentDate = new Date();
+
+    return paginatedDSDM?.map((model) => {
+      const enrolledDate = model.dateEnrolled ? parseDate(model.dateEnrolled) : null;
+      const formattedEnrolledDate = enrolledDate ? formatDate(enrolledDate) : '';
+
+      const completedDate = model.dateCompleted ? parseDate(model.dateCompleted) : currentDate;
+      const formattedCompletedDate = model.dateCompleted ? formatDate(completedDate) : 'Current';
+
+      return {
+        ...model,
+        id: model?.program?.concept?.uuid,
+        model: model?.display,
+        dateEnrolled: formattedEnrolledDate,
+        dateCompleted: formattedCompletedDate,
+      };
+    });
+  }, [paginatedDSDM]);
+
+  if (isLoading) {
+    return <DataTableSkeleton role="progressbar" />;
+  }
+
+  if (paginatedDSDM?.length >= 0) {
```

Comment:
i really don't think u need this if . You can basically just return the datatable since there is already empty array catch